pkg/agent/inteface.go

```diff
@@ -117,6 +118,25 @@ type InstallStrategy struct {
 
 type HealthProber struct {
 	Type HealthProberType
+
+	WorkProber WorkHealthProber
```

Comment:
aesthetically i'm leaning on the typical union struct:

```
type HealthProber struct {
         // Type is the type of health checker
	Type HealthProberType 
         // Work specifies how to map the status of the addon's underlying ManifestWork 
         // resource to the overall healthiness of the addon.
         // Optional if the type is not "Work"
	Work *WorkHealthProber {
             // ProbeFields is a list of resource fields to track from the ManifestWork.
             ProbeFields []ProbeField
             // FieldHealthCheckFunc is a function that asserts the healthiness of the addon.
             // Returning a nil error means the addon is healthy.
             FieldHealthCheckFunc HealthCheck(workapiv1.ResourceIdentifier, workapiv1.StatusFeedbackResult) error
         }
}         
```

Reply:
fixed